src/App.tsx

```diff
@@ -1,26 +1,158 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useMemo, useState } from 'react';
+import { Todo } from './types/Todo';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoFooter } from './components/TodoFooter';
+import { ErrorNotification } from './components/ErrorNotification';
+import { addTodo, deleteTodo, getTodos, USER_ID } from './api/todos';
+import { Error } from './types/Error';
+import { TodoItem } from './components/TodoItem';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<Error | null>(null);
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingIds, setLoadingIds] = useState<number[]>([]);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const data = await getTodos();
+
+        setTodos(data);
+      } catch (error) {
+        setErrorMessage(Error.LoadTodos);
+      }
+    })();
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      if (filter === Filter.All) {
+        return true;
+      }
+
+      return filter === Filter.Completed ? todo.completed : !todo.completed;
+    });
+  }, [todos, filter]);
+
+  const todosCounter: number = useMemo(() => {
+    return todos.filter(todo => !todo.completed).length;
+  }, [todos]);
+
+  const todosCompletedCounter: number = useMemo(() => {
```

Comment:
you can use a common mathematical expression instead of the iterative method, this will ease the resource requirements of the application
`todos.length - todosCompletedCounter`

Reply:
> you can use a common mathematical expression instead of the iterative method, this will ease the resource requirements of the application `todos.length - todosCompletedCounter`

Good point, but up to you